checksum/checksum.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	tc, err := checksumFor(filepath.Join(target, name))
+	if err != nil {
+		return false, fmt.Errorf("error getting target %s checksum: %w", name, err)
+	}
+
+	return bytes.Equal(sc, tc), nil
+}
+
+// Check compares md5 checksum files from src directory with corresponding files in target directory
+func Check(src, target string) error {
+	ls, err := filepath.Glob(filepath.Join(src, "*.md5"))
+	if err != nil {
+		return fmt.Errorf("error listing checksum files in %s directory: %w", src, err)
+	}
+
+	if len(ls) == 0 {
+		return fmt.Errorf("target directory %s has no checksum files to compare with", target)
+	}
+
+	bar := progressbar.Default(int64(len(ls)))
+	defer bar.Close()
+	bar.Describe("Checking files checksum")
+	if err := bar.RenderBlank(); err != nil {
+		return fmt.Errorf("error rendering the progress bar: %w", err)
+	}
+
+	type qm struct {
+		f     string
+		equal bool
+		err   error
+	}
+	q := make(chan qm, len(ls))
+	defer close(q)
+
+	var wg sync.WaitGroup
+
+	for _, f := range ls {
+		wg.Add(1)
+
+		go func(f string, wg *sync.WaitGroup, q chan<- qm) {
+			defer wg.Done()
+
+			equal, err := check(src, target, f)
+
+			q <- qm{f, equal, err}
+		}(f, &wg, q)
+	}
+
+	wg.Wait()
+
+	var notEqual []string
+	for m := range q {
+		if m.err != nil {
+			return err
+		}
+		bar.Add(1)
+		if bar.IsFinished() {
+			if len(notEqual) > 0 {
+				return fmt.Errorf("got different checksum for file(s): %v", notEqual)
+			}
+
+			log.Output(1, "OK!")
+			return nil
+		}
+
+		if !m.equal {
+			notEqual = append(notEqual, m.f)
+		}
+	}
+
+	return nil
+}
+
+func create(src, target string, f fs.DirEntry) error {
+	isHidden := func(f fs.DirEntry) bool {
+		return string(f.Name()[0]) == "."
+	}
+
+	if f.IsDir() || isHidden(f) {
+		return nil
+	}
+
+	checksum, err := checksumFor(filepath.Join(src, f.Name()))
+	if err != nil {
+		return fmt.Errorf("error getting checksum for %s: %w", f.Name(), err)
+	}
+
+	base := strings.TrimSuffix(f.Name(), filepath.Ext(f.Name()))
+	cfn := fmt.Sprintf("%s.md5", base)
+	if err := os.WriteFile(filepath.Join(target, cfn), checksum, 0755); err != nil {
+		return fmt.Errorf("error writing %s checksum file: %w", cfn, err)
+	}
+
+	return nil
+}
+
+// Create creates a md5 checksum file in target directory for each file in src directory
+func Create(src, target string) error {
+	ls, err := os.ReadDir(src)
+	if err != nil {
+		return fmt.Errorf("error reading %s directory: %w", src, err)
+	}
+
+	bar := progressbar.Default(int64(len(ls)))
+	defer bar.Close()
+	bar.Describe("Creating checksum files")
+	if err := bar.RenderBlank(); err != nil {
+		return fmt.Errorf("error rendering the progress bar: %w", err)
+	}
+
+	q := make(chan error, len(ls))
+	defer close(q)
+
+	var wg sync.WaitGroup
+
+	for _, f := range ls {
+		wg.Add(1)
+
+		go func(f fs.DirEntry, wg *sync.WaitGroup, q chan<- error) {
+			defer wg.Done()
+
+			q <- create(src, target, f)
+		}(f, &wg, q)
+	}
+
+	wg.Wait()
+
+	for err := range q {
```

Comment:
Aqui não precisamos do `WaitGrup` se utilizarmos:

```suggestion
	for i := 0; i < len(ls); i++ {
		err := <- q
```